Assets/MixedRealityToolkit.Providers/WindowsMixedReality/XRSDK/Controllers/WindowsMixedRealityXRSDKArticulatedHand.cs

```diff
@@ -0,0 +1,343 @@
+                                triangles = handMeshTriangleIndices,
+                                uvs = handMeshUVs,
+                                position = translation.ToUnityVector3(),
+                                rotation = rotation.ToUnityQuaternion()
+                            };
+
+                            CoreServices.InputSystem?.RaiseHandMeshUpdated(InputSource, ControllerHandedness, handMeshInfo);
+                        }
+                    }
+                }
+            }
+#endif // WINDOWS_UWP
+        }
+
+        private void UpdateIndexFingerData(MixedRealityInteractionMapping interactionMapping)
+        {
+            // Update the interaction data source
+            interactionMapping.PoseData = unityJointPoses[TrackedHandJoint.IndexTip];
+
+            // If our value changed raise it.
+            if (interactionMapping.Changed)
+            {
+                // Raise input system event if it's enabled
+                CoreServices.InputSystem?.RaisePoseInputChanged(InputSource, ControllerHandedness, interactionMapping.MixedRealityInputAction, interactionMapping.PoseData);
+            }
+        }
+
+        /// <summary>
+        /// Converts a Unity finger bone into an MRTK hand joint.
+        /// </summary>
+        /// <remarks>
+        /// For HoloLens 2, Unity provides four joints per finger, in index order of metacarpal (0) to tip (4).
+        /// The first joint for the thumb is the wrist joint. Palm joint is not provided.
+        /// </remarks>
+        /// <param name="finger">The Unity classification of the current finger.</param>
+        /// <param name="index">The Unity index of the current finger bone.</param>
+        /// <returns>The current Unity finger bone converted into an MRTK joint.</returns>
+        private TrackedHandJoint ConvertToTrackedHandJoint(HandFinger finger, int index)
+        {
+            switch (finger)
+            {
+                case HandFinger.Thumb: return (index == 0) ? TrackedHandJoint.Wrist : TrackedHandJoint.ThumbMetacarpalJoint + index - 1;
+                case HandFinger.Index: return TrackedHandJoint.IndexMetacarpal + index;
+                case HandFinger.Middle: return TrackedHandJoint.MiddleMetacarpal + index;
+                case HandFinger.Ring: return TrackedHandJoint.RingMetacarpal + index;
+                case HandFinger.Pinky: return TrackedHandJoint.PinkyMetacarpal + index;
+                default: return TrackedHandJoint.None;
+            }
+        }
+
+        #region IMixedRealityHand Implementation
+
+        /// <inheritdoc/>
+        public bool TryGetJoint(TrackedHandJoint joint, out MixedRealityPose pose) => unityJointPoses.TryGetValue(joint, out pose);
+
+        #endregion IMixedRealityHand Implementation
+
+        private readonly float CursorBeamBackwardTolerance = 0.5f;
+        private readonly float CursorBeamUpTolerance = 0.8f;
+
+        /// <inheritdoc/>
+        public override bool IsInPointingPose
```

Comment:
I know this was merged in, but I don't think this logic should be repeated in many places.

It exists in the hand ray class, and I think it makes the most sense to lean on an implementation for the HandRay, in order to determine if Pointing pose is active. 

This logic exists in each articulated hand implementation, as well is in the hand ray class, and the math is the same everywhere.

Reply:
Yep, I agree! #7146 moved logic that this class and the legacy WMR articulated hand class had into one shared definition class.
In upcoming PRs, I have plans to expand this pattern out to the rest of the articulated hands, to better support #6973 on all platforms, which will also allow for fewer places this logic is defined.